### DISCUSSION TITLE
Auto-generate TOC of pages visible to user?

### DISCUSSION THREAD

**Discussion by JetForMe** [requarks/wiki#3411] [Help / Questions]
**Question**
Is there a way to auto-generate a ToC of pages visible to the currently-logged-in user? It seems the login form doesn't redirect to the original URL after login, but rather to the top level. On that page I'd like to show a ToC for that user, so they can find their way to the original page they were trying to get to.

**Host Info (please complete the following information):**
OS: [e.g. Ubuntu 18.04, Docker]
Wiki.js version: [e.g. 2.0.1]
Database engine: [e.g. postgres 9.7, mysql 5.7]

**Comment by JetForMe** [comment]
I should add: obviously login should remember the user's original destination, but it would be nice to get TOCs for each access group they can go to by clicking the wiki icon in the top-left.

**Comment by e4rthdog** [comment]
I am ending up running python scripts to generate the index pages for my categories. I would love to have this natively.
